src/audio/smart_amp/smart_amp_maxim_dsm.c

```diff
@@ -60,6 +60,162 @@ static int maxim_dsm_init(struct smart_amp_mod_struct_t *hspk, struct comp_dev *
 	return (int)retcode;
 }
 
+static int maxim_dsm_get_all_param(struct smart_amp_mod_struct_t *hspk,
+				   struct comp_dev *dev)
+{
+	struct smart_amp_caldata *caldata = &hspk->param.caldata;
+	int32_t *db = (int32_t *)caldata->data;
+	enum DSM_API_MESSAGE retcode;
+	int cmdblock[1 + DSM_DEFAULT_NUM_CHANNEL];
+	int num_param = hspk->param.max_param;
+	int x;
+
+	for (x = 0 ; x < num_param ;  x++) {
```

Comment:
Can we have an inline comments for this function and one for the loop explaining what it does.

Reply:
Agreed. Shall add more explanation.